thrust/system/detail/generic/shuffle.inl

```diff
@@ -32,6 +32,44 @@ namespace system {
 namespace detail {
 namespace generic {
 
+class round_function {
+private:
+  constexpr static uint64_t _wyp0 = 0xa0761d6478bd642full;
+  constexpr static uint64_t _wyp1 = 0xe7037ed1a0b428dbull;
+  constexpr static uint64_t _wyp2 = 0x8ebc6af09c88c6e3ull;
+  constexpr static uint64_t _wyp3 = 0x589965cc75374cc3ull;
+  constexpr static uint64_t _wyp4 = 0x1d8e4e27c47d124full;
+
+#ifdef __CUDA_ARCH__
```

Comment:
Is this the correct method of switching behaviour for host vs device implementations